### DISCUSSION TITLE
Does MediaMTX remove SPS from H.264 video to make an SDP instead?

### DISCUSSION THREAD

**Discussion by rfolland** [bluenviron/mediamtx#2672] [Questions]
### Question

Does MediaMTX actually remove SPS from H.264-streams sent in from ffmpeg, to make an SDP instead? If so, is it possible to keep the SPS in the video stream? And repeat it regularly? This makes it possible to read the stream without using the SDP. I am using RTSP.

**Comment by aler9** [comment]
> Does MediaMTX actually remove SPS from H.264-streams sent in from ffmpeg, to make an SDP instead? If so, is it possible to keep the SPS in the video stream? And repeat it regularly? This makes it possible to read the stream without using the SDP. I am using RTSP.

Hello, in case of RTSP the video stream is left untouched, while the SDP is updated when the SPS/PPS in the stream changes.
Regarding other protocols, SPS/PPS are not removed, on the contrary, they are always added before every IDR.

**Reply by rfolland** [reply]
OK, thank you for confirming. That is what I would have expected. Then I should be able to get this to work as planned. Thanks!

**Comment by rfolland** [comment]
Hi again, I am still having problems with this. I have tried feeding MediaMTX with ffmpeg (with -f rtsp) and gstreamer (with rtspclientsink), but both end up with streaming from MediaMTX with SDP and no SPS/PPS in the actual videostreams. 

Should I feed MedaMTX in a different way to keep the videostreams intact?

**Reply by aler9** [reply]
Hello, MediaMTX doesn't remove SPS/PPS in any way. If you still have problems, open an issue and provide a network dump by following instructions.

[a comment by an automated account is not shown]
